server/app/services/applicant/ApplicantService.java

```diff
@@ -495,41 +564,124 @@ private void notifyTiSubmitter(
+        getPreferredTILocale(tiEmail).toCompletableFuture();
+    return localeFuture
+        .thenRunAsync(
+            () -> {
+              // Not blocking since it already completed
+              Locale locale = localeFuture.join().orElse(LocalizedStrings.DEFAULT_LOCALE);
+              Messages messages =
+                  messagesApi.preferred(ImmutableSet.of(Lang.forCode(locale.toLanguageTag())));
+              String subject =
+                  messages.at(
+                      MessageKey.EMAIL_TI_APPLICATION_SUBMITTED_SUBJECT.getKeyName(),
+                      programName,
+                      applicantId);
+              String message =
+                  String.format(
+                      "%s\n%s",
+                      messages.at(
+                          MessageKey.EMAIL_TI_APPLICATION_SUBMITTED_BODY.getKeyName(),
+                          programName,
+                          applicantId,
+                          applicationId),
+                      messages.at(
+                          MessageKey.EMAIL_TI_MANAGE_YOUR_CLIENTS.getKeyName(), tiDashLink));
+              if (isStaging) {
+                amazonSESClient.send(stagingTiNotificationMailingList, subject, message);
+              } else {
+                amazonSESClient.send(tiEmail, subject, message);
+              }
+            })
+        .toCompletableFuture();
   }
 
-  private void maybeNotifyApplicant(long applicantId, long applicationId, String programName) {
-    Optional<String> email = getEmail(applicantId).toCompletableFuture().join();
-    if (email.isEmpty()) {
-      return;
-    }
-    String civiformLink = baseUrl;
-    String subject = String.format("Your application to program %s is received", programName);
-    String message =
-        String.format(
-            "Your application to program %s has been received. Your applicant ID is %d and the"
-                + " application ID is %d.\n"
-                + "Log in to CiviForm at %s.",
-            programName, applicantId, applicationId, civiformLink);
-    if (isStaging) {
-      amazonSESClient.send(stagingApplicantNotificationMailingList, subject, message);
-    } else {
-      amazonSESClient.send(email.get(), subject, message);
-    }
+  private CompletionStage<Void> maybeNotifyApplicant(
+      long applicantId, long applicationId, String programName) {
+    CompletableFuture<Optional<String>> emailFuture = getEmail(applicantId).toCompletableFuture();
+    CompletableFuture<Optional<Locale>> localeFuture =
+        getPreferredLocale(applicantId).toCompletableFuture();
+    return CompletableFuture.allOf(emailFuture, localeFuture)
+        .thenRunAsync(
+            () -> {
+              // Not blocking since these are already completed
+              Optional<String> email = emailFuture.join();
+              Locale locale = localeFuture.join().orElse(LocalizedStrings.DEFAULT_LOCALE);
+              if (!email.isEmpty()) {
```

Comment:
**nitpick:** invert the condition and return early instead to avoid indenting the code currently in the `if` body